node/lib/util/synthetic_gc_util.js

```diff
@@ -112,14 +156,29 @@ SyntheticGcUtil.prototype.getBareSubmoduleRepo = co.wrap(
         const fetcher = new SubmoduleFetcher(repo, refHeadCommit);
 
         let subUrl = yield fetcher.getSubmoduleUrl(subName);
+        let subPath = subUrl;
 
-        const metaUrl = yield GitUtil.getOriginUrl(repo);
-        if (metaUrl !== null) {
-            subUrl = SubmoduleConfigUtil.resolveSubmoduleUrl(metaUrl,
-                                                             subUrl);
+        if (subUrl.startsWith(".")) {
+            // relative url specified
+            subUrl = detail.rebuildUrl(subUrl);
+            subPath = yield SyntheticBranchUtil.urlToLocalPath(repo, subUrl);
+        }
+
+        if (!fs.existsSync(subPath)) {
+            console.log("Submodule at following path does not exists: " +
+                subPath);
+
+            const metaUrl = yield GitUtil.getOriginUrl(repo);
+            if (metaUrl !== null) {
```

Comment:
When would this be null, and do we really want to not fetch in that case?  Should we just fail loudly here?